### PR TITLE
[1.17] Update CNI to v0.8.6

### PR DESCRIPTION
**Pull request by justaugustus**
**What type of PR is this?**

/kind cleanup
/area dependency release-eng
/sig network release

**What this PR does / why we need it**:

Update CNI to v0.8.6

cc: @kubernetes/release-engineering @kubernetes/sig-network-misc 
/assign @BenTheElder @yujuhong @tallclair
ref: https://github.com/kubernetes/release/pull/1309

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
- Update CNI to v0.8.6
- build: Use debian-hyperkube-base@v1.0.0 image
```

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by cblecker** [comment]
/hold
Requesting details on why this is being picked back

File: build/dependencies.yaml [hunk 1]
```diff
@@ -1,13 +1,7 @@
...
   - name: "cni"
-    version: 0.7.5
+    version: 0.8.6
     refPaths:
-    - path: build/debian-hyperkube-base/Makefile
-      match: CNI_VERSION=
-    - path: build/rpms/kubeadm.spec
-      match: kubernetes-cni
-    - path: build/rpms/kubelet.spec
-      match: kubernetes-cni
```
Comment (liggitt) [inline build/dependencies.yaml:10]: same question about dropping these refs without actually removing those files

**Comment by justaugustus** [comment]
Image has been promoted: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/post-k8sio-cip/1265499696156119040
/hold cancel

File: cluster/gce/util.sh [hunk 2]
```diff
@@ -1136,7 +1136,8 @@ NODE_PROBLEM_DETECTOR_VERSION: $(yaml-quote ${NODE_PROBLEM_DETECTOR_VERSION:-})
 NODE_PROBLEM_DETECTOR_TAR_HASH: $(yaml-quote ${NODE_PROBLEM_DETECTOR_TAR_HASH:-})
 NODE_PROBLEM_DETECTOR_RELEASE_PATH: $(yaml-quote ${NODE_PROBLEM_DETECTOR_RELEASE_PATH:-})
 NODE_PROBLEM_DETECTOR_CUSTOM_FLAGS: $(yaml-quote ${NODE_PROBLEM_DETECTOR_CUSTOM_FLAGS:-})
-CNI_STORAGE_PATH: $(yaml-quote ${CNI_STORAGE_PATH:-})
+CNI_STORAGE_URL_BASE: $(yaml-quote ${CNI_STORAGE_URL_BASE:-})
+CNI_TAR_PREFIX: $(yaml-quote ${CNI_TAR_PREFIX:-})
```
Comment (liggitt) [inline cluster/gce/util.sh:1140]: @bowei to review cluster CNI changes

File: cluster/gce/config-default.sh [hunk 3]
```diff
@@ -301,9 +301,9 @@ NODE_PROBLEM_DETECTOR_TAR_HASH="${NODE_PROBLEM_DETECTOR_TAR_HASH:-}"
 NODE_PROBLEM_DETECTOR_RELEASE_PATH="${NODE_PROBLEM_DETECTOR_RELEASE_PATH:-}"
 NODE_PROBLEM_DETECTOR_CUSTOM_FLAGS="${NODE_PROBLEM_DETECTOR_CUSTOM_FLAGS:-}"
 
-CNI_STORAGE_PATH="${CNI_STORAGE_PATH:-https://storage.googleapis.com/kubernetes-release/network-plugins}"
-CNI_VERSION="${CNI_VERSION:-}"
-CNI_SHA1="${CNI_SHA1:-}"
+CNI_SHA1=${CNI_SHA1:-}
+CNI_TAR_PREFIX=${CNI_TAR_PREFIX:-cni-plugins-linux-amd64-}
+CNI_STORAGE_URL_BASE=${CNI_STORAGE_URL_BASE:-https://storage.googleapis.com/k8s-artifacts-cni/release}
```
Comment (liggitt) [inline cluster/gce/config-default.sh:306]: @bowei to review

**Comment by liggitt** [comment]
bazel and build bits look reasonable, tagged the network team to review the cluster CNI changes

**Comment by bowei** [comment]
CNI changes 

/approve

**Pull request merged**